src/CoreWCF.Primitives/src/CoreWCF/Security/SecuritySessionSecurityTokenAuthenticator.cs

```diff
@@ -1127,12 +1136,14 @@ internal ChannelDispatcher InitializeRuntime(SecurityServiceDispatcher securityD
                 //  }
 
                 //Replacing above code by below 3 lines.(adding securityservicedispatcher to demuxer, how to respond)
+
                 Binding binding = _authenticator.IssuerBindingContext.Binding;
-                binding.ReceiveTimeout = _authenticator.NegotiationTimeout;
-                securityDispatcher.ChannelBuilder.AddServiceDispatcher<IReplyChannel>(securityDispatcher, new ChannelDemuxerFilter(contractFilter, filterPriority));
+                // binding.ReceiveTimeout = _authenticator.NegotiationTimeout;
+                binding.ReceiveTimeout = TimeSpan.FromMinutes(20); // for testing
+                securityDispatcher.ChannelBuilder.AddServiceDispatcher<TChannel>(securityDispatcher, new ChannelDemuxerFilter(contractFilter, filterPriority));
```

Comment:
Remove this. You can modify this in the test. Take a look at some of the client binding helpers where I detect if a debugger is attached and crank up the timeouts really high. You should do something like this with the passed in bindings to CoreWCF.

Reply:
done..